pkgs/by-name/we/weasis/package.nix

```
@@ -0,0 +1,85 @@
+{ lib
+, stdenv
+, fetchzip
+, jre
+, copyDesktopItems
+, makeDesktopItem
+}:
+
+let
+  # Build process uses this naming for output
+  platform = if stdenv.isLinux then "linux"
+             else if stdenv.isDarwin then "macosx"
+             else if stdenv.isWindows then "windows"
+             else throw "unsupported platform";
+  arch = if stdenv.isx86_64 then "x86-64"
+         else if stdenv.isAarch64 then "aarch64"
+         else if stdenv.isArmv7a then "armv7a"
+         else throw "unsupported arch";
+
+in stdenv.mkDerivation rec {
+  pname = "weasis";
+  version = "4.2.1";
+
+  # Their build instructions indicate to use the packaging script
+  src = fetchzip {
+    url = "https://github.com/nroduit/Weasis/releases/download/v${version}/weasis-native.zip";
+    hash = "sha256-HDlylpe8cHZRaIXndfGh6XmUn8o2PQB1Av7hLCp679U=";
+    stripRoot = false;
+  };
+
+  nativeBuildInputs = [
+    copyDesktopItems
+  ];
+
+  desktopItems = [
+    (makeDesktopItem {
+      name = "DICOMizer";
+      exec = "Dicomizer";
+      icon = "Dicomizer";
+      desktopName = "DICOMizer";
+      comment = "Convert standard images into DICOM";
+    })
+    (makeDesktopItem {
+      name = "Weasis";
+      exec = "Weasis";
+      icon = "Weasis";
+      desktopName = "Weasis";
+      comment = meta.description;
+    })
+  ];
+
+  postPatch = ''
+    patchShebangs ./build/script/package-weasis.sh
+  '';
+
+  buildPhase = ''
+    runHook preBuild
+
+    ./build/script/package-weasis.sh --no-installer --jdk ${jre}
+
+    runHook postBuild
+  '';
+
+  installPhase = ''
+    runHook preInstall
+
+    mkdir -p $out/share/{applications,pixmaps}
+
+    mv weasis-${platform}-${arch}-jdk${lib.versions.major jre.version}-${version}/Weasis/* $out/
+    mv $out/lib/*.png $out/share/pixmaps/
+
+    runHook postInstall
+  '';
+
+  meta = {
+    description = "Multipurpose standalone and web-based DICOM viewer with a highly modular architecture";
+    homepage = "https://weasis.org";
+    # Using changelog from releases as it is more accurate
+    changelog = "https://github.com/nroduit/Weasis/releases/tag/v${version}";
+    license = with lib.licenses; [ asl20 epl20 ];
+    maintainers = [ lib.maintainers.wolfangaukang ];
+    platforms = [ "x86_64-linux" ];
```

Why do you map multiple platforms (`linux`, `macos`, `windows`) and archs (`x86_64`, `aarch64`, `arm`) at the start of this derivation,  to only support `x86_64-linux`?

In this case, it is better to just support just `x86_64-linux`. If someone later wants to support the other platforms they can add support for it later.

---

Ok, the last phrase kind of solves my dilemma then.

---

BTW, a cool trick when you want to support only those platforms you declare in the `platform` attrset:

```suggestion
    platforms = lib.attrNames platform;
```

You will need to remove the `throwSystem` part, but there is no problem because the meta validation will automatically throw it for you.